csrc/scheduler/vectorize_helper.cpp

```diff
@@ -64,63 +64,96 @@ ContiguousInnerDimensionsMapper::ContiguousInnerDimensionsMapper(
+  //                   id) != reference->getMaybeRFactorDomain().end());
+  //         }),
+  //     "\nIterDomains passed in to ContiguousInnerDimensionsMapper passed in
+  //     to ", "ContiguousInnerDimensionsMapper must all exist in the rfactor
+  //     domain.\n", "Reference: ", reference->toString());
+
+  // // Record while processing reference's information
+  // recording_ = true;
+  // std::shared_ptr<Information> reference_information;
+
+  // // Ordering of dimensions is important in this analysis, if an ordering is
+  // // contiguous in the reference, but not the target tensor views, then we
+  // // cannot consider that a contiguous merge dimension for vectorization.
+  // std::vector<IterDomain*> reordered_rfactor;
+  // for (auto id : reference->getMaybeRFactorDomain()) {
+  //   // Exclude reduction IDs if the reference is a fusion input as they
+  //   // don't manifest at all in the fusion. This simplifies the
+  //   // analysis in getContigMergeOfInnerSize, which only looks at
+  //   // non-reduction rfactor domains. Including reduction domains here
+  //   // can result in incorrect ordering
+  //   if (reference->isFusionInput() && id->isReduction()) {
+  //     continue;
+  //   }
+  //   // If not a fusion input, can this be a reduction domain?
+  //   TORCH_INTERNAL_ASSERT(
+  //       !id->isReduction(),
+  //       "Unexpected reduction domain: ",
+  //       id->toString(),
+  //       " of tensor ",
+  //       reference->toString());
+  //   if (std::find(reference_ids.begin(), reference_ids.end(), id) !=
+  //       reference_ids.end()) {
+  //     reordered_rfactor.push_back(id);
+  //     // Initiailze the extent for the mapped iter domain
+  //     addProjectedExtent(id, commonOrConstExtent(ca_map_, id));
+  //   } else if (!id->isBroadcast()) {
+  //     // Ignore broadcasts in the reference. Otherwise, remove non-contiguous
+  //     // IDs in the reference tensor as this is the contiguous mapper.
+  //     reordered_rfactor.clear();
+  //   }
+  // }
   // Record while processing reference's information
   recording_ = true;
-  std::shared_ptr<Information> reference_information;
-
-  // Ordering of dimensions is important in this analysis, if an ordering is
-  // contiguous in the reference, but not the target tensor views, then we
-  // cannot consider that a contiguous merge dimension for vectorization.
-  std::vector<IterDomain*> reordered_rfactor;
-  for (auto id : reference->getMaybeRFactorDomain()) {
-    // Exclude reduction IDs if the reference is a fusion input as they
-    // don't manifest at all in the fusion. This simplifies the
-    // analysis in getContigMergeOfInnerSize, which only looks at
-    // non-reduction rfactor domains. Including reduction domains here
-    // can result in incorrect ordering
-    if (reference->isFusionInput() && id->isReduction()) {
-      continue;
+
+  auto projected_rfactor = projectId(ids, reference->getMaybeRFactorDomain());
```

Comment:
Should we `addProjectedExtent` for `ids` before projecting to rfactor?

Reply:
I think this isn't what the old behavior required. having said that, sounds like a good idea to add some extra bits there?!

I'm actually seeing another failure related to no extend. `RNGTest.BroadcastingRNGSmem`. Let me try this and also take another look at that failing test.